cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java

```diff
@@ -122,7 +124,7 @@ private void addTicket(final Ticket ticket, final long ttl) {
      */
     @Override
     public Ticket getTicket(final String ticketId) {
-        return this.registry.get(ticketId);
+        return getProxiedTicketInstance(this.registry.get(ticketId));
```

Comment:
@leleuj pointed out that ticket update is done by proxy object. I analyzed it and found out that the root cause of ticket not being updated in case of hazelcast registry was that it wasn't returning proxy object, while other registries do.